sceptre/template_handlers/http.py

```diff
@@ -66,18 +77,57 @@ def handle(self):
 
         return template
 
-    def _get_template(self, url):
+    def _get_template(self, url, retries, timeout):
         """
         Get template from the web
         :param url: The url to the template
         :type: str
-        :returns: The body of the CloudFormation template.
-        :rtype: str
+        :returns: The number of retry attempts.
+        :rtype: int
+        :returns: The timeout for the session in seconds.
+        :rtype: int
         """
         self.logger.debug("Downloading file from: %s", url)
+        session = self._get_retry_session(retries=retries)
         try:
-            response = requests.get(url)
+            response = session.get(url, timeout=timeout)
             return response.content
-        except requests.exceptions.RequestException as e:
-            self.logger.fatal(e)
+        except (InvalidURL, ConnectTimeout) as e:
             raise e
+
+    def _get_retry_session(self,
+                           retries,
+                           backoff_factor=0.3,
+                           status_forcelist=(429, 500, 502, 503, 504),
+                           session=None):
+        """
+        Get a request session with retries.  Retry options are explained in the request libraries
+        https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#module-urllib3.util.retry
+        """
+        session = session or requests.Session()
+        retry = Retry(
+            total=retries,
+            read=retries,
+            connect=retries,
+            backoff_factor=backoff_factor,
+            status_forcelist=status_forcelist,
+        )
+        adapter = HTTPAdapter(max_retries=retry)
+        session.mount('http://', adapter)
+        session.mount('https://', adapter)
+        return session
+
+    def _get_handler_option(self, name, default):
+        """
+        Get the template handler options
+        :param url: The option name
+        :type: str
+        :param default: The default value if option is not set.
+        :rtype: int
+        """
+        if HANDLER_OPTION_KEY in self.stack_group_config:
+            option = self.stack_group_config.get(HANDLER_OPTION_KEY)
+            if name in option:
+                return option.get(name)
```

Comment:
Why aren't these argument options? I'm confused as to why you're pulling these in via the stack_group_config rather than arguments defined in the handler's schema.

Reply:
It's because the template schema is defined per sceptre config.   I think  it makes more sense to apply these options at a higher level, across all sceptre configs.  That's why I've opted to pull from stack_group_config.

Reply:
I see... then I think this should definitely be included in the docs. Otherwise, how would anyone know about these settings?

Reply:
good call, I completely forgot.  It's there now.  Thanks!